src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -212,8 +213,13 @@ private async Task<bool> InvokeConfigLoadedHandlersAsync()
         return results.All(x => x);
     }
 
-    private static RuntimeConfig HandleCosmosNoSqlConfiguration(string? schema, RuntimeConfig runtimeConfig, string connectionString)
+    private static RuntimeConfig HandleCosmosNoSqlConfiguration(string? schema, RuntimeConfig runtimeConfig, string connectionString, string? databaseName = null)
     {
+        if (string.IsNullOrEmpty(databaseName))
```

Comment:
Lets keep this argument name as dataSourceName since this is different than the databasename in the connection string

Reply:
fixed.